src/lib/operators/table_scan/column_vs_column_table_scan_impl.cpp

```diff
@@ -28,66 +29,99 @@ std::string ColumnVsColumnTableScanImpl::description() const { return "ColumnVsC
 
 std::shared_ptr<PosList> ColumnVsColumnTableScanImpl::scan_chunk(ChunkID chunk_id) const {
   const auto chunk = _in_table->get_chunk(chunk_id);
+  const auto& left_segment = *chunk->get_segment(_left_column_id);
+  const auto& right_segment = *chunk->get_segment(_right_column_id);
+
+  std::shared_ptr<PosList> result;
+
+  // If the left and the right segment and/or type are not the same, we erase the types even for the release build.
+  // For example, ValueSegment<int> == ValueSegment<float> will be erased. So will ValueSegment<int> ==
+  // DictionarySegment<int>. ReferenceSegments do not need to be handled differently because we expect a table to
+  // either have only ReferenceSegments or non-ReferenceSegments.
+  //
+  // We use type erasure here because we currently do not compare, e.g., a ValueSegment with a DictionarySegment,
```

Comment:
"we currently do not" could be read as "Hyrise cannot do this". What is actually meant, is that we have no use cases where this happens. I'd make that clearer.

Reply:
like this?

Reply:
nah, at first I didn't even see a difference. "actively use", maybe?